### ISSUE TITLE
Find a better way to force non-desktop bindings even when switching applications

### ISSUE THREAD

**Issue by Alia5** [Alia5/GloSC#1]
Currently, a bug in Steam, using a console window, is used to stay in shortcut bindings even if applications get switched.

I tried looking at what Steam does but my reversing skills are very very basic.
So far I've seen that Steam checks the current ForegroundWindow using GetForegroundWindow, I tried hooking that, but had no luck.

**Comment by Alia5** [comment]
I might have found a better way:

In steamclient.dll the AppID of the (to become) active controller-binding gets moved to the EAX register
By doing a mid function hook, injecting a little bit of code, every available binding can get enforced.

Here's a signature ```"\x8B\x45\x0c\x57\x8B\x7D\x08"``` (all bytes matched) so you can easy find the address

**Comment by MJavad** [comment]
Do you have any ideas about improving this? I may be able to help in RE.

**Comment by DawsSangio** [comment]
Hi, just to want to share some interesting experience. I've found that the minecraft launcher (java pc) keep hooking even when switching applicatins. Just add it as non steam game, disable use desktop shortcut launcer, and it still keep any type of bindings, xinput and km. It should be interesting to find which kind of code it use just to  replicate a common target, avoiding any type of memory hack.

**Comment by Alia5** [comment]
@MJavad: No idea on how to do this **properly** yet.

@DawsSangio: You mean without GloSC? Well for me that's not the case. Still switches happily configuration... (Also I doubt Minecraft hooks anything :D )

**Comment by DawsSangio** [comment]
Yes, without GloSC. Launching the new official Minecraft Java frontend let Steam hook and remain with custom bindings even when loosing focus.
I've tested with several different pc with success, indeed I'm using this trick to play the UWP version of Minecraft with dedicated KM setup.

![minecraft_launcer_0](https://user-images.githubusercontent.com/26057951/28248569-e4d88758-6a46-11e7-978d-b8beadca9db4.jpg)

![minecraft_launcer_1](https://user-images.githubusercontent.com/26057951/28248570-e89443e6-6a46-11e7-9d8c-a485e1120ddc.jpg)

**Comment by Alia5** [comment]
As I've said, doesn't work for me.
